src/tests/unit/tests/background/dev-tools-monitor.test.ts

```diff
@@ -0,0 +1,237 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { OnDevToolStatusPayload } from 'background/actions/action-payloads';
+import { DevToolsMonitor } from 'background/dev-tools-monitor';
+import { TabContextManager } from 'background/tab-context-manager';
+import { BrowserAdapter } from 'common/browser-adapters/browser-adapter';
+import { Messages } from 'common/messages';
+import {
+    DelayCreator,
+    PromiseFactory,
+    TimeoutCreator,
+    TimeoutError,
+} from 'common/promises/promise-factory';
+import _ from 'lodash';
+import { flushSettledPromises } from 'tests/common/flush-settled-promises';
+import { IMock, It, Mock, Times } from 'typemoq';
+import { DictionaryNumberTo } from 'types/common-types';
+
+class TestDevToolsMonitor extends DevToolsMonitor {
+    public activeDevtoolTabIds: number[];
+
+    public async testPollLoop(): Promise<void> {
+        await super.monitorActiveDevtools();
```

Comment:
Ah, I see now why this has to be protected... I don't love that we can't keep this private but I don't see an alternative if we want to have good test coverage. :/

Reply:
The main reason is to give us a way to await the polling loop and/or catch exceptions in tests, since the public function is intentionally non-asynchronous. I agree that it's awkward, but I also couldn't think of another way to do this without pushing the unawaited promise somewhere else where the intention might be less clear. I'm open to alternatives if there are any suggestions!

Reply:
Makes sense, I can't think of a better way either so this looks fine.